### ISSUE TITLE
BlitzCache divide by zero

### ISSUE THREAD

**Issue by BlitzErik** [BrentOzarULTD/SQL-Server-First-Responder-Kit#1747]
Reported here: https://dba.stackexchange.com/questions/216823/sp-blitzcache-gives-a-divide-by-zero-error-encountered

Fix should be easy enough.

**Comment by RichBenner** [comment]
god dammit, just spent an hour narrowing this down myself only to find it's a known bug, at least it's not just me :)

**Comment by BrentOzar** [comment]
Think of it this way: you have verified your skills as a top-notch code investigator.

**Comment by BlitzErik** [comment]
@RichBenner since you did all that work, do you want the glory of adding the fix?

**Comment by RichBenner** [comment]
I'm happy to, I used CASE statements to fix it for my application, is this the approach you'd take in the actual scripts?

**Comment by BlitzErik** [comment]
@RichBenner yep, totally fine!

**Comment by EvanCarroll** [comment]
@RichBenner generally the trick to making this more painless when working with awesome Microsoft stuff is to subtract the line offset from the crap until the procedure starting point and start looking at the offset for the next few statements which may be hundreds of lines, but still. It shouldn't take that long.

**Comment by RichBenner** [comment]
@EvanCarroll yeah, that's the way I debug this stuff too. The difficulty here was that I was logging sp_BlitzFirst to a table and it was at that point that I was getting errors. I had to dig down to find that it was BlitzCache throwing errors before then narrowing down the section of the code that was causing issues. I then replicated the issue and dug into the data to find exactly the plans that were causing my issue. Hence, an hour :)
